src/petals/server/handler.py

```diff
@@ -234,6 +234,59 @@ async def rpc_inference(
                         prefix_length += length_increment
             finally:
                 self._log_request("rpc_inference.close", requested_uids, context)
+                if todo_use_async_with is not None:
+                    await todo_use_async_with.__aexit__(None, None, None)  # todo: change into async with
+
+    @contextlib.asynccontextmanager
+    async def _managed_session(self, session_id: str):
+        assert session_id not in self._session_queues, f"session id {session_id} is not unique"
+        try:
+            self._session_queues[session_id] = asyncio.Queue()
+            self._session_handlers[session_id] = self._handler_index
+            for other_index, other_queue in enumerate(self._handler_queues):
+                if other_index != self._handler_index:
+                    other_queue.put_nowait((Event.NEW_SESSION, session_id, self._handler_index))
+            yield
+        finally:
+            await self._session_queues.pop(session_id).put(None)  # put None so that the get task will not hang
+            del self._session_handlers[session_id]
+            for other_index, other_queue in enumerate(self._handler_queues):
+                if other_index != self._handler_index:
+                    other_queue.put_nowait((Event.END_SESSION, session_id, self._handler_index))
+
+    async def _put_into_push_queue(self, session_id: str, request: runtime_pb2.ExpertRequest):
+        handler_index = self._session_handlers.get(session_id)
+        if handler_index is None:
+            logger.debug(f"Ignored rpc_push to unknown session ID: {session_id}")
+        elif handler_index == self._handler_index:
+            await self._session_queues[session_id].put(request)
```

Comment:
```suggestion
            self._session_queues[session_id].put_nowait(request)
```

Thus, this functon doesn't need to be `async` - please make it a usual func

Reply:
+